channeldb/migration14/enclosed_codec.go

```diff
@@ -0,0 +1,312 @@
+package migration14
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/lnwire"
```

Comment:
One drive-by comment: here we are still relying on packages that may change and break the migration. These are basic types, so it is unlikely, but the migration isn't fully self-contained like for example a sql migration script would be or migration 13.

Reply:
The main package here that impacts the migration is `lnwire`, specifically the way we write wire messages to disk. I don't think this will be changing any time soon. If they attributes or types of any of the imported packages change, then if they materially affect the migration, this PR should no longer compile.

Reply:
Especially the wire format is something that isn't set in stone, but of course it depends on what our support period is for this migration. Also a future change may be detected because the system doesn't build anymore, but that does mean that more copying needs to happen at that point. Touching an old migration to make it build again is a risk. I favor the fully isolated low-level approach.

Reply:
Copying is essentially zero cost, it's all effectively dead code anyway. As mentioned offline, making `migration14` into a module should resolve any issues with the referenced types/packages changing under us (so it should always build given the deps are static).